### PR TITLE
chore: refactor CPU monitor for cgroups

### PR DESCRIPTION
**Pull request by davidzhao**

### CODE REVIEW

**Comment by davidzhao** [comment]
@martin-purplefish no major functionality change from your PR, just moved code around. I made CPU count a float, since it's possible to give a docker pod a fractional CPU

**Comment by martin-purplefish** [comment]
Nice this looks very clean!

File: livekit-agents/livekit/agents/utils/__init__.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 from livekit import rtc
 
-from . import aio, audio, codecs, http_context, images
+from . import aio, audio, codecs, http_context, hw, images
 from .audio import AudioBuffer, combine_frames, merge_frames
 from .exp_filter import ExpFilter
+from .hw import get_cpu_monitor
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/utils/__init__.py:6]: let's not export this function, ppl can get it from `hw`

**Review by theomonnom — APPROVED** [review]
lgtm

**Pull request merged**